### PR TITLE
Od 16276 add discount dsl

### PR DESCRIPTION
**Pull request by ArtyomKravchenko**

### CODE REVIEW

**Review by ari — CHANGES_REQUESTED** [review]
Great, could we also have the same for the other end of the relationship from Discount. And the remove... options.

File: server/src/main/groovy/ish/oncourse/server/cayenne/CourseClass.groovy [hunk 1]
```diff
@@ -199,9 +199,9 @@ class CourseClass extends _CourseClass implements CourseClassTrait, ICourseClass
 	}
 
 	/**
-	 * This is a faked flattened relationship via discountCourseClasses.discount.
-	 *
-	 * @return the discounts for this class
+	 * Return the immutable list of discounts available for class
+	 * To add more available discounts for the class see {@link ish.oncourse.server.cayenne.CourseClassTrait#addDiscount(Discount)}
```
Comment (ari) [inline server/src/main/groovy/ish/oncourse/server/cayenne/CourseClass.groovy:203]: I don't think this @link will work in our API docs publishing process.

Comment (ArtyomKravchenko) [reply server/src/main/groovy/ish/oncourse/server/cayenne/CourseClass.groovy]: fixed

**Comment by ArtyomKravchenko** [comment]
I have added remove API as well. Also added similar API for discount entity (from other end)

**Review by rzenov — APPROVED** [review]
Reviewed

**Review by ari — APPROVED** [review]

**Pull request merged**